feat(arithmetic): Adding a parser for arithmetic

wmak opened the pull request:
- This is just a simple arithmetic parser that can handle just numbers
  for now
- This is so that later it can be extended to handle variables to create
  more complicated Discover columns

wmak committed:
feat(arithmetic): Adding a parser for arithmetic

- This is just a simple arithmetic parser that can handle just numbers
  for now
- This is so that later it can be extended to handle variables to create
  more complicated Discover columns

Zylphrex reviewed (approved):
Just a small comment. But otherwise, LGTM

wmak committed:
ref: Simplify spaces in the grammar

wmak commented on src/sentry/discover/arithmetic.py:
```diff
@@ -0,0 +1,142 @@
+add_sub_operator     = spaces (plus / minus) spaces
+mul_div_operator     = spaces (multiply / divide) spaces
+# TODO(wmak) allow variables
+primary              = numeric_value
```
:+1: good idea, made the visitors for all three the same too which is nice

k-fish reviewed (approved):
Arithmetic 😮 .

What happens with nested parens and order of operations? Is there a test for it? (should there be?)

wmak commented:
@k-fish so this should have basic order of operations (ie. multiplication & division have higher priortity than addition and subtraction) but not brackets **yet** (ie. DMAS but not BEDMAS) and is currently roughly tested by
https://github.com/getsentry/sentry/pull/26057/files#diff-a56d3d53717fef131973ee39a0a7a85999414265d37573db0a137b381cc4a3dbR83-R91

I'll add a few more, its just that I can't do the same amount of parametrization as the other tests since i have to craft the asserts by hand

wmak committed:
ref: Fixing a bug with the flatten method

- Wasn't actually flattening so `1+2+3+4` just became `1+2+4`
- Fixing the repr so its easier test
